lib/src/tasks/coverage/api.dart

```diff
@@ -568,11 +550,9 @@ class CoverageTask extends Task {
           break;
         }
       }
-      if (server != null && server.observatoryPort.length > 0) {
-        List<int> result = await server.checkPorts();
-        return result;
-      }
-      return [port];
+      var observatoryPorts = await SeleniumHelper.getActiveObservatoryPorts();
```

Comment:
Since the test/coverage tasks are no longer responsible for starting the selenium server, the API for getting the active observatory ports is now statically available. This is temporary and would be replaced by the plugin system I mentioned before, at which point the test/coverage tasks would be completely decoupled from the selenium dependency.